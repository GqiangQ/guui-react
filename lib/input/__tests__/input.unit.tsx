// import {mount} from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Input from '../input';
import {mount} from 'enzyme';

describe(Input.displayName, () => {
  it('DisplayName', () => {
    expect(Input.displayName).toEqual('Input');
  });
  it('prop label = hello', () => {
    const tree = renderer
      .create(<Input label="hello"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('prop labelPosition = top', () => {
    const tree = renderer
      .create(<Input label="hello" labelPosition="top"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('prop error = wrong', () => {
    const tree = renderer
      .create(<Input error="wrong"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('prop errorPosition = bottom', () => {
    const tree = renderer
      .create(<Input error="wrong" errorPosition="bottom"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('prop size = small', () => {
    const tree = renderer
      .create(<Input size="small"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('prop size = big', () => {
    const tree = renderer
      .create(<Input size="big"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('prop length = 10', () => {
    const tree = renderer
      .create(<Input length={10}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('prop value = 123', () => {
    const tree = renderer
      .create(<Input value="123"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should update value', () => {
    const c = mount(<Input value="234" onChange={() => {c.setProps({value: '233'});}}/>);
    expect(c.find('input').prop('value')).toEqual('234');
    c.find('input').simulate('change');
    expect(c.find('input').prop('value')).toEqual('233');
  });
  it('should response to enter', () => {
    const fn = jest.fn();
    const c = mount(<Input onEnter={fn}/>);
    c.find('input').simulate('keydown', {keyCode: 13});
    expect(fn.mock.calls.length).toEqual(1);
  });
  it('should response to keydown', () => {
    const fn = jest.fn();
    const c = mount(<Input onKeyDown={fn}/>);
    c.find('input').simulate('keydown', {keyCode: 42});
    expect(fn.mock.calls.length).toEqual(1);
  });
});
