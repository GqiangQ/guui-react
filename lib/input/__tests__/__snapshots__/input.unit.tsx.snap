// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input prop error = wrong 1`] = `
<div
  className="gu-input-wrapper"
>
  <div
    className="gu-input-inputAndError"
  >
    <input
      className="gu-input gu-input-hasError "
      onKeyDown={[Function]}
      spellCheck={false}
      style={
        Object {
          "width": undefined,
        }
      }
    />
    <span
      className="gu-input-error"
    >
      wrong
    </span>
  </div>
</div>
`;

exports[`Input prop errorPosition = bottom 1`] = `
<div
  className="gu-input-wrapper"
>
  <div
    className="gu-input-inputAndError gu-input-errorBottom"
  >
    <input
      className="gu-input gu-input-hasError "
      onKeyDown={[Function]}
      spellCheck={false}
      style={
        Object {
          "width": undefined,
        }
      }
    />
    <span
      className="gu-input-error"
    >
      wrong
    </span>
  </div>
</div>
`;

exports[`Input prop label = hello 1`] = `
<div
  className="gu-input-wrapper"
>
  <label
    className="gu-input-label"
  >
    hello
  </label>
  <div
    className="gu-input-inputAndError"
  >
    <input
      className="gu-input "
      onKeyDown={[Function]}
      spellCheck={false}
      style={
        Object {
          "width": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Input prop labelPosition = top 1`] = `
<div
  className="gu-input-wrapper gu-input-labelTop"
>
  <label
    className="gu-input-label"
  >
    hello
  </label>
  <div
    className="gu-input-inputAndError"
  >
    <input
      className="gu-input "
      onKeyDown={[Function]}
      spellCheck={false}
      style={
        Object {
          "width": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Input prop length = 10 1`] = `
<div
  className="gu-input-wrapper"
>
  <div
    className="gu-input-inputAndError"
  >
    <input
      className="gu-input "
      onKeyDown={[Function]}
      spellCheck={false}
      style={
        Object {
          "width": "calc(10em + 18px)",
        }
      }
    />
  </div>
</div>
`;

exports[`Input prop size = big 1`] = `
<div
  className="gu-input-wrapper"
>
  <div
    className="gu-input-inputAndError"
  >
    <input
      className="gu-input gu-input-big "
      onKeyDown={[Function]}
      spellCheck={false}
      style={
        Object {
          "width": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Input prop size = small 1`] = `
<div
  className="gu-input-wrapper"
>
  <div
    className="gu-input-inputAndError"
  >
    <input
      className="gu-input gu-input-small "
      onKeyDown={[Function]}
      spellCheck={false}
      style={
        Object {
          "width": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Input prop value = 123 1`] = `
<div
  className="gu-input-wrapper"
>
  <div
    className="gu-input-inputAndError"
  >
    <input
      className="gu-input "
      onKeyDown={[Function]}
      spellCheck={false}
      style={
        Object {
          "width": undefined,
        }
      }
      value="123"
    />
  </div>
</div>
`;
